runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java

```diff
@@ -60,68 +61,131 @@
+  public static final String COMBINE_GLOBALLY_TRANSFORM_URN = "beam:transform:combine_globally:v1";
+  public static final String RESHUFFLE_URN = "beam:transform:reshuffle:v1";
+  public static final String WRITE_FILES_TRANSFORM_URN = "beam:transform:write_files:v1";
+
+  // CombineComponents
   public static final String COMBINE_PER_KEY_PRECOMBINE_TRANSFORM_URN =
-      getUrn(CombineComponents.COMBINE_PER_KEY_PRECOMBINE);
+      "beam:transform:combine_per_key_precombine:v1";
   public static final String COMBINE_PER_KEY_MERGE_ACCUMULATORS_TRANSFORM_URN =
-      getUrn(CombineComponents.COMBINE_PER_KEY_MERGE_ACCUMULATORS);
+      "beam:transform:combine_per_key_merge_accumulators:v1";
   public static final String COMBINE_PER_KEY_EXTRACT_OUTPUTS_TRANSFORM_URN =
-      getUrn(CombineComponents.COMBINE_PER_KEY_EXTRACT_OUTPUTS);
-  public static final String RESHUFFLE_URN = getUrn(StandardPTransforms.Composites.RESHUFFLE);
-  public static final String WRITE_FILES_TRANSFORM_URN =
-      getUrn(StandardPTransforms.Composites.WRITE_FILES);
+      "beam:transform:combine_per_key_extract_outputs:v1";
+  public static final String COMBINE_GROUPED_VALUES_TRANSFORM_URN =
+      "beam:transform:combine_grouped_values:v1";
 
   // SplittableParDoComponents
   public static final String SPLITTABLE_PAIR_WITH_RESTRICTION_URN =
-      getUrn(SplittableParDoComponents.PAIR_WITH_RESTRICTION);
+      "beam:transform:sdf_pair_with_restriction:v1";
   public static final String SPLITTABLE_SPLIT_RESTRICTION_URN =
-      getUrn(SplittableParDoComponents.SPLIT_RESTRICTION);
+      "beam:transform:sdf_split_restriction:v1";
+  /**
+   * @deprecated runners should move away from using `SplittableProcessKeyedElements` and prefer to
+   *     internalize any necessary SplittableDoFn expansion.
+   */
+  @Deprecated
   public static final String SPLITTABLE_PROCESS_KEYED_URN =
-      getUrn(SplittableParDoComponents.PROCESS_KEYED_ELEMENTS);
+      "beam:transform:sdf_process_keyed_elements:v1";
+
   public static final String SPLITTABLE_PROCESS_ELEMENTS_URN =
-      getUrn(SplittableParDoComponents.PROCESS_ELEMENTS);
+      "beam:transform:sdf_process_elements:v1";
   public static final String SPLITTABLE_SPLIT_AND_SIZE_RESTRICTIONS_URN =
-      getUrn(SplittableParDoComponents.SPLIT_AND_SIZE_RESTRICTIONS);
+      "beam:transform:sdf_split_and_size_restrictions:v1";
   public static final String SPLITTABLE_PROCESS_SIZED_ELEMENTS_AND_RESTRICTIONS_URN =
-      getUrn(SplittableParDoComponents.PROCESS_SIZED_ELEMENTS_AND_RESTRICTIONS);
-
-  public static final String ITERABLE_SIDE_INPUT =
-      getUrn(RunnerApi.StandardSideInputTypes.Enum.ITERABLE);
-  public static final String MULTIMAP_SIDE_INPUT =
-      getUrn(RunnerApi.StandardSideInputTypes.Enum.MULTIMAP);
+      "beam:transform:sdf_process_sized_element_and_restrictions:v1";
+
+  static {
+    checkState(PAR_DO_TRANSFORM_URN.equals(getUrn(StandardPTransforms.Primitives.PAR_DO)));
```

Comment:
Add comment: // Primitives ？(same as line 67)

Reply:
I'll include that in a followup PR where I change many of the existing getUrn calls as you suggest.